dbms/src/Columns/ColumnString.cpp

```diff
@@ -460,6 +481,301 @@ void ColumnString::getPermutationWithCollationImpl(
     }
 }
 
+void ColumnString::countSerializeByteSize(PaddedPODArray<size_t> & byte_size) const
+{
+    RUNTIME_CHECK_MSG(byte_size.size() == size(), "size of byte_size({}) != column size({})", byte_size.size(), size());
+
+    if unlikely (!offsets.empty() && offsets.back() > UINT32_MAX)
+    {
+        size_t sz = size();
+        for (size_t i = 0; i < sz; ++i)
+            RUNTIME_CHECK_MSG(
+                sizeAt(i) > UINT32_MAX,
```

Comment:
Looks like you want to find the first string whose length is greater than `UINT32_MAX`?
Bug if `sizeAt(i) > UINT32_MAX` is true, for loop will iter to next value.

Reply:
Good catch! Fixed.